### ISSUE TITLE
Disconnection Issue with tripplite_usb

### ISSUE THREAD

**Issue by Jsalas424** [networkupstools/nut#2001]
My tripplite_usb connected UPS keeps disconnecting and reconnecting rapidly throwing a bunch of errors into my logs and interfering with my monitoring tool [nut_exporter](https://github.com/DRuggeri/nut_exporter). 

Relevant Syslog Output: 
```
Jul 30 11:44:46 Server tripplite_usb[2745772]: Successfully reconnected
Jul 30 11:44:46 Server tripplite_usb[2745772]: Using binary SMART protocol (3005)
Jul 30 11:44:47 Server tripplite_usb[2745772]: Unit ID: 65535
Jul 30 11:46:16 Server tripplite_usb[2745772]: libusb_set_report() returned 0 instead of 8
Jul 30 11:46:16 Server tripplite_usb[2745772]: Error reading 'V' value: Device detached? (error 0: error sending control message: Broken pipe)
Jul 30 11:46:16 Server tripplite_usb[2745772]: Reconnect attempt #1
Jul 30 11:46:16 Server tripplite_usb[2745772]: Successfully reconnected
Jul 30 11:46:16 Server tripplite_usb[2745772]: Using binary SMART protocol (3005)
Jul 30 11:46:17 Server tripplite_usb[2745772]: Unit ID: 65535
Jul 30 11:46:46 Server tripplite_usb[2745772]: libusb_set_report() returned 0 instead of 8
Jul 30 11:46:46 Server tripplite_usb[2745772]: Error reading 'P' value: Device detached? (error 0: error sending control message: Broken pipe)
Jul 30 11:46:46 Server tripplite_usb[2745772]: Reconnect attempt #1
Jul 30 11:46:47 Server tripplite_usb[2745772]: Successfully reconnected
Jul 30 11:46:47 Server tripplite_usb[2745772]: Using binary SMART protocol (3005)
Jul 30 11:46:48 Server tripplite_usb[2745772]: Unit ID: 65535
Jul 30 11:46:58 Server tripplite_usb[2745772]: libusb_set_report() returned 0 instead of 8
Jul 30 11:46:58 Server tripplite_usb[2745772]: Error reading L value: Device detached? (error 0: error sending control message: Broken pipe)
Jul 30 11:46:58 Server tripplite_usb[2745772]: Reconnect attempt #1
Jul 30 11:46:58 Server upsd[2745774]: Data for UPS [TrippLiteUPS] is stale - check driver
Jul 30 11:46:59 Server tripplite_usb[2745772]: Successfully reconnected
Jul 30 11:46:59 Server tripplite_usb[2745772]: Using binary SMART protocol (3005)
Jul 30 11:47:00 Server tripplite_usb[2745772]: Unit ID: 65535
Jul 30 11:47:01 Server upsd[2745774]: UPS [TrippLiteUPS] data is no longer stale
Jul 30 11:47:05 Server tripplite_usb[2745772]: libusb_set_report() returned 0 instead of 8
Jul 30 11:47:05 Server tripplite_usb[2745772]: Error reading M (min/max input voltage): Device detached? (error 0: error sending control message: Broken pipe)
Jul 30 11:47:05 Server tripplite_usb[2745772]: Reconnect attempt #1
Jul 30 11:47:05 Server upsd[2745774]: Data for UPS [TrippLiteUPS] is stale - check driver
Jul 30 11:47:05 Server tripplite_usb[2745772]: Successfully reconnected
Jul 30 11:47:05 Server tripplite_usb[2745772]: Using binary SMART protocol (3005)
Jul 30 11:47:06 Server tripplite_usb[2745772]: Unit ID: 65535
Jul 30 11:47:07 Server upsd[2745774]: UPS [TrippLiteUPS] data is no longer stale
```

Here's some debug data where you can see that 1 second in we get connectivity issues followed by reconnection at 9 seconds. 
```
root@Server:~# /lib/nut/tripplite_usb -a TrippLiteUPS  -DDDDDD
Network UPS Tools - Tripp Lite OMNIVS / SMARTPRO driver 0.29 (2.7.4)
Warning: This is an experimental driver.
Some features may not function correctly.

   0.000000     debug level is '6'
   0.164905     Checking device (8087/8000) (003/002)
   0.233076     - VendorID: 8087
   0.233093     - ProductID: 8000
   0.233099     - Manufacturer: unknown
   0.233113     - Product: unknown
   0.233117     - Serial Number: unknown
   0.233131     - Bus: 003
   0.233135     - Device release number: 0004
   0.233139     Trying to match device
   0.233145     Device does not match - skipping
   0.233175     Checking device (1D6B/0002) (003/001)
   0.233230     - VendorID: 1d6b
   0.233243     - ProductID: 0002
   0.233246     - Manufacturer: unknown
   0.233249     - Product: unknown
   0.233264     - Serial Number: unknown
   0.233267     - Bus: 003
   0.233270     - Device release number: 0515
   0.233283     Trying to match device
   0.233287     Device does not match - skipping
   0.233304     Checking device (1D6B/0003) (004/001)
   0.264061     - VendorID: 1d6b
   0.264074     - ProductID: 0003
   0.264086     - Manufacturer: unknown
   0.264088     - Product: unknown
   0.264091     - Serial Number: unknown
   0.264105     - Bus: 004
   0.264109     - Device release number: 0515
   0.264114     Trying to match device
   0.264121     Device does not match - skipping
   0.264133     Checking device (09AE/0001) (002/005)
   0.267628     - VendorID: 09ae
   0.267634     - ProductID: 0001
   0.267639     - Manufacturer: Tripp Lite 
   0.267644     - Product: TRIPP LITE SMART500RT1U 
   0.267649     - Serial Number: unknown
   0.267653     - Bus: 002
   0.267658     - Device release number: 000a
   0.267662     Trying to match device
   0.267667     Device matches
   0.267675     failed to claim USB device: could not claim interface 0: Device or resource busy
   0.267711     detached kernel driver from USB device...
   0.267726     nut_usb_set_altinterface: skipped usb_set_altinterface(udev, 0)
   0.267732     Detected a UPS: Tripp Lite /TRIPP LITE SMART500RT1U 
   0.267740     send_to_all: SETINFO ups.vendorid "09ae"
   0.267746     send_to_all: SETINFO ups.productid "0001"
   0.267752     send_to_all: SETINFO device.type "ups"
   0.267758     send_to_all: SETINFO driver.version "2.7.4"
   0.267764     send_to_all: SETINFO driver.version.internal "0.29"
   0.267769     send_to_all: SETINFO driver.name "tripplite_usb"
   0.267774     send_cmd(msg_len=2, type='
   0.267780     send_cmd: sending  3a 00 ff 0d 00 00 00 00 '........'
   0.267785     send_cmd send_try 1
   1.372280     libusb_get_interrupt: Connection timed out
   1.372298     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   2.372555     libusb_get_interrupt: Connection timed out
   2.372589     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   3.372970     libusb_get_interrupt: Connection timed out
   3.372989     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   4.373400     libusb_get_interrupt: Connection timed out
   4.373418     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   5.373810     libusb_get_interrupt: Connection timed out
   5.373828     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   6.374780     libusb_get_interrupt: Connection timed out
   6.374797     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   7.375979     libusb_get_interrupt: Connection timed out
   7.375997     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   8.377059     libusb_get_interrupt: Connection timed out
   8.377076     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   9.377960     libusb_get_interrupt: Connection timed out
   9.378011     libusb_get_interrupt() returned 0 instead of 8 while sending 3a 00 ff 0d 00 00 00 00 '........'
   9.378016     send_cmd send_try 2
   9.483798     send_cmd: received 00 30 05 58 58 58 58 0d '.0.XXXX.' (OK)
   9.483843     send_cmd: send_try = 2, recv_try = 1

   9.483857     Using binary SMART protocol (3005)
   9.483873     send_to_all: SETINFO ups.firmware.aux "protocol 3005"
   9.483878     send_cmd(msg_len=3, type='W')
   9.483882     send_cmd: sending  3a 57 00 a8 0d 00 00 00 '.W......'
   9.483885     send_cmd send_try 1
   9.587778     send_cmd: received 57 00 0d 00 00 00 00 00 'W.......' (OK)
   9.587828     send_cmd: send_try = 1, recv_try = 1

   9.587833     send_cmd(msg_len=2, type='S')
   9.587847     send_cmd: sending  3a 53 ac 0d 00 00 00 00 '.S......'
   9.587850     send_cmd send_try 1
   9.691783     send_cmd: received 53 01 04 00 00 64 00 0d 'S....d..' (OK)
   9.691802     send_cmd: send_try = 1, recv_try = 1

   9.691824     send_to_all: SETINFO ups.mfr "Tripp Lite"
   9.691829     send_cmd(msg_len=2, type='P')
   9.691835     send_cmd: sending  3a 50 af 0d 00 00 00 00 '.P......'
   9.691840     send_cmd send_try 1
   9.795777     send_cmd: received 50 30 30 35 30 30 58 0d 'P00500X.' (OK)
   9.795797     send_cmd: send_try = 1, recv_try = 1

   9.795827     send_to_all: SETINFO ups.model "TRIPP LITE SMART500RT1U"
   9.795835     send_to_all: SETINFO ups.power.nominal "500"
   9.795842     send_cmd(msg_len=2, type='F')
   9.795850     send_cmd: sending  3a 46 b9 0d 00 00 00 00 '.F......'
   9.795855     send_cmd send_try 1
   9.899762     send_cmd: received 46 32 39 41 46 20 42 0d 'F29AF.B.' (OK)
   9.899784     send_cmd: send_try = 1, recv_try = 1

   9.899802     send_to_all: SETINFO ups.firmware "F29AF.B"
   9.899816     send_cmd(msg_len=2, type='V')
   9.899823     send_cmd: sending  3a 56 a9 0d 00 00 00 00 '.V......'
   9.899828     send_cmd send_try 1
  10.003784     send_cmd: received 56 02 00 0c 01 58 58 0d 'V....XX.' (OK)
  10.003822     send_cmd: send_try = 1, recv_try = 1

  10.003827     Switchable load banks: 1
  10.003834     send_cmd(msg_len=2, type='V')
  10.003840     send_cmd: sending  3a 56 a9 0d 00 00 00 00 '.V......'
  10.003845     send_cmd send_try 1
  10.107772     send_cmd: received 56 02 00 0c 01 58 58 0d 'V....XX.' (OK)
  10.107787     send_cmd: send_try = 1, recv_try = 1

  10.107814     send_to_all: SETINFO ups.debug.V "02 00 0c 01 58 58 0d '....XX.'"
  10.107821     send_to_all: SETINFO outlet.1.id "1"
  10.107834     send_to_all: SETINFO outlet.1.desc "Load 1"
  10.107838     send_to_all: SETINFO outlet.1.switchable "1"
  10.107857     send_to_all: SETINFO outlet.1.switch "1"
  10.107863     send_to_all: SETFLAGS outlet.1.switch RW STRING
  10.107867     send_to_all: SETAUX outlet.1.switch 3
  10.107871     send_to_all: SETINFO outlet.2.id "2"
  10.107876     send_to_all: SETINFO outlet.2.desc "Load 2"
  10.107882     send_to_all: SETINFO outlet.2.switchable "0"
  10.107886     send_to_all: ADDCMD load.on
  10.107890     send_to_all: ADDCMD load.off
  10.107894     send_cmd(msg_len=2, type='U')
  10.107899     send_cmd: sending  3a 55 aa 0d 00 00 00 00 '.U......'
  10.107903     send_cmd send_try 1
  10.211779     send_cmd: received 55 ff ff 0d 00 00 00 00 'U.......' (OK)
  10.211796     send_cmd: send_try = 1, recv_try = 1

  10.211816     send_to_all: SETINFO ups.id "65535"
  10.211822     send_to_all: SETFLAGS ups.id RW STRING
  10.211828     send_to_all: SETAUX ups.id 5
  10.211833     Unit ID: 65535
  10.211840     send_to_all: SETINFO input.voltage.nominal "120"
  10.211844     send_to_all: SETINFO battery.voltage.nominal "12"
  10.211849     send_to_all: SETINFO ups.debug.load_banks "1"
  10.211854     send_to_all: SETINFO ups.delay.shutdown "64"
  10.211860     send_to_all: SETFLAGS ups.delay.shutdown RW STRING
  10.211865     send_to_all: SETAUX ups.delay.shutdown 3
  10.211870     send_to_all: ADDCMD test.battery.start
  10.211875     send_to_all: ADDCMD reset.input.minmax
  10.211879     send_to_all: ADDCMD shutdown.return
Attached to Tripp Lite TRIPP LITE SMART500RT1U
  10.211889     send_cmd(msg_len=2, type='S')
  10.211897     send_cmd: sending  3a 53 ac 0d 00 00 00 00 '.S......'
  10.211902     send_cmd send_try 1
  10.315772     send_cmd: received 53 01 04 00 00 64 00 0d 'S....d..' (OK)
  10.315788     send_cmd: send_try = 1, recv_try = 1

  10.315800     send_to_all: SETINFO ups.debug.S "01 04 00 00 64 00 0d '....d..'"
  10.315815     send_to_all: SETINFO ups.status "OL"
  10.315820     send_cmd(msg_len=2, type='D')
  10.315835     send_cmd: sending  3a 44 bb 0d 00 00 00 00 '.D......'
  10.315839     send_cmd send_try 1
  10.419812     send_cmd: received 44 00 74 00 89 0d 00 00 'D.t.....' (OK)
  10.419829     send_cmd: send_try = 1, recv_try = 1

  10.419838     send_to_all: SETINFO input.voltage "116"
  10.419846     send_to_all: SETINFO battery.voltage "13.70"
  10.419851     send_cmd(msg_len=2, type='M')
  10.419857     send_cmd: sending  3a 4d b2 0d 00 00 00 00 '.M......'
  10.419861     send_cmd send_try 1
  10.523786     send_cmd: received 4d 00 57 00 7c 0d 00 00 'M.W.....' (OK)
  10.523816     send_cmd: send_try = 1, recv_try = 1

  10.523830     send_to_all: SETINFO input.voltage.minimum " 87"
  10.523834     send_to_all: SETINFO input.voltage.maximum "124"
  10.523837     send_cmd(msg_len=2, type='T')
  10.523841     send_cmd: sending  3a 54 ab 0d 00 00 00 00 '.T......'
  10.523843     send_cmd send_try 1
^C
```

Config of `/etc/nut/ups.conf`:
```
maxretry = 3

[TrippLiteUPS]
        driver = tripplite_usb
        port = auto
        desc = "Rack Mounted Tripp Lite"
        pollinterval = 1
```

Added `DEADTIME 25` and  `MAXAGE 25` to `/etc/nut/upsmon.conf`

`/etc/nut/nut.conf`:
```
MODE=netserver
```

`/etc/nut/upsmon.conf`:
```
MINSUPPLIES 1
SHUTDOWNCMD "/sbin/shutdown -h +0"
POLLFREQ 5
POLLFREQALERT 5
HOSTSYNC 15

DEADTIME 25
MAXAGE 25

POWERDOWNFLAG /etc/killpower
RBWARNTIME 43200
NOCOMMWARNTIME 300
FINALDELAY 5

MONITOR TrippLiteUPS@localhost 1 username password master
```

I've seen USB cables implicated in this, I'm using a brand new StarTech USB A to B cable so I don't think it's that. 

I see someone suggesting [adding new `udev` rules ](https://serverfault.com/questions/994338/how-can-i-fix-my-ups-repeatedly-disconnecting-and-reconnecting) or adding `pollonly = "enabled"` to deal with this, is this still necessary?

**Comment by jimklimov** [comment]
Hello, is this some problem that started recently?

How long do you have/monitor this device?

Were there any other recent hardware changes (can we implicate e.g. USB cabling and/or noise on it - you do mention a new cable) or software ones (something else trying to kidnap the libusb device)?

Are you sure one instance of the driver is running; in case of restart - that the older instance did die (and no service/script/crontab resurrects it in a separate process tree)?

CC: @dzomaya @aquette

**Comment by Jsalas424** [comment]
I will certainly look into all these factors, in the meantime I'm dropping off some new interesting logs:

```
Jul 30 20:36:48 Server tripplite_usb[2778403]: Successfully reconnected
Jul 30 20:36:48 Server tripplite_usb[2778403]: Using binary SMART protocol (3005)
Jul 30 20:36:48 Server tripplite_usb[2778403]: libusb_set_report() returned 0 instead of 8
Jul 30 20:36:48 Server tripplite_usb[2778403]: Could not reset watchdog. Please send model information to nut-upsdev mailing list
Jul 30 20:36:49 Server tripplite_usb[2778403]: libusb_get_interrupt() returned 0 instead of 8 while sending 3a 53 ac 0d 00 00 00 00 '.S......'
Jul 30 20:36:50 Server tripplite_usb[2778403]: libusb_get_interrupt() returned 0 instead of 8 while sending 3a 53 ac 0d 00 00 00 00 '.S......'
Jul 30 20:36:51 Server tripplite_usb[2778403]: libusb_get_interrupt() returned 0 instead of 8 while sending 3a 53 ac 0d 00 00 00 00 '.S......'
```

Data has been flowing okay for the past couple of hours until ~10min ago when it suddenly stopped altogether and now all I'm getting is `Jul 30 20:49:11 Server upsmon[2778364]: Poll UPS [TrippLiteUPS@localhost] failed - Driver not connected`

<img width="709" alt="image" src="https://github.com/networkupstools/nut/assets/51519895/39f08602-04a9-460e-bbd8-98b917ff413a">

**Comment by Jsalas424** [comment]
> Were there any other recent hardware changes (can we implicate e.g. USB cabling and/or noise on it - you do mention a new cable) or software ones (something else trying to kidnap the libusb device)?

No other hardware changes. This is running directly on my proxmox hypervisor, this is one of the only software's installed on top of the base OS. I checked every VM to make sure I wasn't inadvertently passing through any USB ports, but I don't pass through anything at all. 

> Are you sure one instance of the driver is running; in case of restart - that the older instance did die (and no service/script/crontab resurrects it in a separate process tree from e.g. systemd unit lifecycle)?

There are no cron jobs for the user

```
root@Server:~# crontab -l
no crontab for root
```

The systemwide cron jobs don't mention anything NUT related:

```
root@Server:~# cat /etc/crontab /etc/cron.d/*
# /etc/crontab: system-wide crontab
# Unlike any other crontab you don't have to run the `crontab'
# command to install the new version when you edit this file
# and files in /etc/cron.d. These files also have username fields,
# that none of the other crontabs do.

SHELL=/bin/sh
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin

# Example of job definition:
# .---------------- minute (0 - 59)
# |  .------------- hour (0 - 23)
# |  |  .---------- day of month (1 - 31)
# |  |  |  .------- month (1 - 12) OR jan,feb,mar,apr ...
# |  |  |  |  .---- day of week (0 - 6) (Sunday=0 or 7) OR sun,mon,tue,wed,thu,fri,sat
# |  |  |  |  |
# *  *  *  *  * user-name command to be executed
17 *    * * *   root    cd / && run-parts --report /etc/cron.hourly
25 6    * * *   root    test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.daily )
47 6    * * 7   root    test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly )
52 6    1 * *   root    test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.monthly )
#
30 3 * * 0 root test -e /run/systemd/system || SERVICE_MODE=1 /usr/lib/x86_64-linux-gnu/e2fsprogs/e2scrub_all_cron
10 3 * * * root test -e /run/systemd/system || SERVICE_MODE=1 /sbin/e2scrub_all -A -r
# cluster wide vzdump cron schedule
# Automatically generated file - do not edit

PATH="/usr/sbin:/usr/bin:/sbin:/bin"

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin

# TRIM the first Sunday of every month.
24 0 1-7 * * root if [ $(date +\%w) -eq 0 ] && [ -x /usr/lib/zfs-linux/trim ]; then /usr/lib/zfs-linux/trim; fi

# Scrub the second Sunday of every month.
24 0 8-14 * * root if [ $(date +\%w) -eq 0 ] && [ -x /usr/lib/zfs-linux/scrub ]; then /usr/lib/zfs-linux/scrub; fi
```



It's not the systemd timers:
```
root@Server:~# systemctl list-timers
NEXT                        LEFT          LAST                        PASSED       UNIT                         ACTIVATES
Wed 2023-08-02 18:54:50 EDT 3min 29s left Wed 2023-08-02 18:49:50 EDT 1min 30s ago smart_exporter.timer         smart_exporter.service
Wed 2023-08-02 22:32:57 EDT 3h 41min left Tue 2023-08-01 22:32:57 EDT 20h ago      systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service
Thu 2023-08-03 00:00:00 EDT 5h 8min left  Wed 2023-08-02 00:00:57 EDT 18h ago      logrotate.timer              logrotate.service
Thu 2023-08-03 00:00:00 EDT 5h 8min left  Wed 2023-08-02 00:00:57 EDT 18h ago      man-db.timer                 man-db.service
Thu 2023-08-03 00:25:39 EDT 5h 34min left Wed 2023-08-02 11:27:46 EDT 7h ago       apt-daily.timer              apt-daily.service
Thu 2023-08-03 05:54:00 EDT 11h left      Wed 2023-08-02 05:38:46 EDT 13h ago      pve-daily-update.timer       pve-dai>
Thu 2023-08-03 06:27:43 EDT 11h left      Wed 2023-08-02 06:45:46 EDT 12h ago      apt-daily-upgrade.timer      apt-dai>
Sun 2023-08-06 03:10:21 EDT 3 days left   Sun 2023-07-30 03:11:09 EDT 3 days ago   e2scrub_all.timer            e2scrub>
Mon 2023-08-07 01:26:07 EDT 4 days left   Mon 2023-07-31 00:33:57 EDT 2 days ago   fstrim.timer                 fstrim.>

9 timers listed.
Pass --all to see loaded but inactive timers, too.
```

The nut-client config:

```
root@Server:~# systemctl status nut-client
● nut-monitor.service - Network UPS Tools - power device monitor and shutdown controller
     Loaded: loaded (/lib/systemd/system/nut-monitor.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-08-02 18:56:42 EDT; 4s ago
    Process: 3142243 ExecStart=/sbin/upsmon (code=exited, status=0/SUCCESS)
   Main PID: 3142245 (upsmon)
      Tasks: 3 (limit: 38361)
     Memory: 924.0K
        CPU: 6ms
     CGroup: /system.slice/nut-monitor.service
             ├─3142244 /lib/nut/upsmon
             └─3142245 /lib/nut/upsmon

Aug 02 18:56:42 TracheNodeB upsmon[3142243]: fopen /run/nut/upsmon.pid: No such file or directory
Aug 02 18:56:42 TracheNodeB upsmon[3142243]: Using power down flag file /etc/killpower
Aug 02 18:56:42 TracheNodeB upsmon[3142243]: UPS: TrippLiteUPS@localhost (master) (power value 1)
Aug 02 18:56:42 TracheNodeB upsmon[3142244]: Startup successful
Aug 02 18:56:42 TracheNodeB systemd[1]: nut-monitor.service: Can't open PID file /run/nut/upsmon.pid (yet?) after start>
Aug 02 18:56:42 TracheNodeB systemd[1]: nut-monitor.service: Supervising process 3142245 which is not our child. We'll >
Aug 02 18:56:42 TracheNodeB systemd[1]: Started Network UPS Tools - power device monitor and shutdown controller.

root@Server:~# cat /lib/systemd/system/nut-monitor.service
[Unit]
Description=Network UPS Tools - power device monitor and shutdown controller
After=local-fs.target network.target nut-server.service

[Service]
ExecStart=/sbin/upsmon
PIDFile=/run/nut/upsmon.pid
Type=forking

[Install]
WantedBy=multi-user.target
```

Nut-server:

```
● nut-server.service - Network UPS Tools - power devices information server
     Loaded: loaded (/lib/systemd/system/nut-server.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2023-07-30 22:16:22 EDT; 2 days ago
    Process: 2385 ExecStart=/sbin/upsd (code=exited, status=0/SUCCESS)
   Main PID: 2386 (upsd)
      Tasks: 1 (limit: 38361)
     Memory: 1.2M
        CPU: 12.515s
     CGroup: /system.slice/nut-server.service
             └─2386 /lib/nut/upsd

root@Server:~# cat /lib/systemd/system/nut-server.service
[Unit]
Description=Network UPS Tools - power devices information server
After=local-fs.target network.target nut-driver.service
# We don't Require drivers to be successfully started! This would be
# a change of behavior compared to init SysV, and could prevent from
# accessing successfully started, at least to audit a system.
Wants=nut-driver.service
Before=nut-monitor.service

[Service]
ExecStart=/sbin/upsd 
Type=forking

[Install]
WantedBy=multi-user.target
```

**Comment by Jsalas424** [comment]
Resolved with upgrade to 2.8.0. 
<img width="704" alt="Screen Shot 2023-11-01 at 4 35 03 PM" src="https://github.com/networkupstools/nut/assets/51519895/66add761-d9ce-4df6-ae2c-ac2650ccd644">

There are still occasional disconnects but it's able to come back online now unlike in v2.7.4. 

```
Nov 01 16:33:02 Server upsd[1092085]: Data for UPS [TrippLiteUPS] is stale - check driver
Nov 01 16:33:03 Server nut-monitor[1092077]: Poll UPS [TrippLiteUPS@localhost] failed - Data stale
Nov 01 16:33:03 Server kernel: usb 3-6: new low-speed USB device number 70 using xhci_hcd
Nov 01 16:33:03 Server kernel: usb 3-6: New USB device found, idVendor=09ae, idProduct=0001, bcdDevice= 0.0a
Nov 01 16:33:03 Server kernel: usb 3-6: New USB device strings: Mfr=1, Product=2, SerialNumber=0
Nov 01 16:33:03 Server kernel: usb 3-6: Product: TRIPP LITE SMART500RT1U 
Nov 01 16:33:03 Server kernel: usb 3-6: Manufacturer: Tripp Lite 
Nov 01 16:33:03 Server kernel: hid-generic 0003:09AE:0001.0140: hiddev0,hidraw0: USB HID v1.10 Device [Tripp Lite  TRIPP LITE SMART500RT1U ] on usb-0000:00:14.0-6/input0
Nov 01 16:33:06 Server tripplite_usb[1921583]: libusb_set_report() returned 0 instead of 8
Nov 01 16:33:06 Server tripplite_usb[1921583]: Error reading S value: Device detached? (error 0: Success)
Nov 01 16:33:06 Server tripplite_usb[1921583]: Reconnect attempt #4
Nov 01 16:33:06 Server tripplite_usb[1921583]: Successfully reconnected
Nov 01 16:33:07 Server tripplite_usb[1921583]: Using binary SMART protocol (3005)
Nov 01 16:33:07 Server tripplite_usb[1921583]: Unit ID: 65535
Nov 01 16:33:08 Server nut-monitor[1092077]: Poll UPS [TrippLiteUPS@localhost] failed - Data stale
Nov 01 16:33:12 Server nut-server[1092085]: UPS [TrippLiteUPS] data is no longer stale
Nov 01 16:33:12 Server upsd[1092085]: UPS [TrippLiteUPS] data is no longer stale
Nov 01 16:33:13 Server nut-monitor[1092077]: Communications with UPS TrippLiteUPS@localhost established
Nov 01 16:33:13 Server nut-monitor[1921710]: Network UPS Tools upsmon 2.8.0
```

The only other interesting point is that I'm getting the following error:
`libusb1: Could not open any HID devices: insufficient permissions on everything`

But that's a fight for another day.

**Comment by andsontan** [comment]
“Could not open any HID devices: insufficient permissions on everything” Have you solved this problem yet? I'm also troubled by it.
